### PR TITLE
Bugfix in retry (assert x ) & add some simple features

### PR DESCRIPTION
**Pull request by nische**

### CODE REVIEW

File: litequeue.py [hunk 1]
```diff
@@ -482,15 +497,18 @@ def full(self) -> bool:
         else:
             return False
 
-    def prune(self):
+    def prune(self, include_failed: bool = True):
         """
         Delete `done` and `failed` messages. # TODO: maybe not failed ones
```
Comment (polyrand) [inline litequeue.py]: The docstring should be updated to reflect the new arguments.

```suggestion
        Delete `DONE` messages.
        
        If `include_failed` is True, the messages in `FAILED` state will be deleted too.
```

File: litequeue.py [hunk 2]
```diff
@@ -421,6 +421,22 @@ def list_locked(self, threshold_seconds: int) -> Iterable[Message]:
         for result in cursor:
             yield Message(**result)
 
+    def list_failed(self) -> Iterable[Message]:
+        """
+        Return all the tasks that have been in the `FAILED`.
```
Comment (polyrand) [inline litequeue.py]:
```suggestion
        Return all the tasks in `FAILED` state.
```

**Review by polyrand — CHANGES_REQUESTED** [review]
Thank you for the PR, this is a nice addition! Just left a couple of change requests related to the docstrings. Everything else looks, good. The PR can be merged after the docstrings are fixed.

**Comment by nische** [comment]
i hope i accepted the changes.

**Review by polyrand — APPROVED** [review]
LGTM!

**Comment by nische** [comment]
I missed to increase the version. I think thats important for the update process on pip?

**Pull request merged**